Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Issue.hs

```diff
@@ -31,84 +47,152 @@ toCommonIssueStatus = \case
+        }
+    mkIssueReportComment :: DC.Comment -> Common.IssueReportCommentItem
+    mkIssueReportComment comment =
+      Common.IssueReportCommentItem
+        { comment = comment.comment,
+          authorDetail = mkAuthorDetails comment.authorId,
+          timestamp = comment.createdAt
+        }
 
 issueUpdate :: ShortId DM.Merchant -> Id DIR.IssueReport -> Common.IssueUpdateReq -> Flow APISuccess
 issueUpdate _merchantShortId issueReportId Common.IssueUpdateReq {..} = do
-  case (status, assignee) of
-    (Just justStatus, Just justAssignee) -> do
-      Esq.runTransaction $ QIR.updateStatusAssignee issueReportId (toDomainIssueStatus justStatus) justAssignee
-      pure Success
-    (Just justStatus, _) -> do
-      Esq.runTransaction $ QIR.updateStatus issueReportId (toDomainIssueStatus justStatus)
-      pure Success
-    (_, Just justAssignee) -> do
-      Esq.runTransaction $ QIR.updateAssignee issueReportId justAssignee
-      pure Success
-    (Nothing, Nothing) -> throwError $ InvalidRequest "Empty request, no fields to update."
-
-issueAddComment :: ShortId DM.Merchant -> Id DIR.IssueReport -> Common.IssueAddCommentReq -> Flow APISuccess
-issueAddComment _merchantShortId issueReportId Common.IssueAddCommentReq {..} = do
-  Esq.runTransaction $ QC.create =<< mkComment
-  pure Success
+  case userId of
+    Just justUserId -> do
+      void $ Esq.runInReplica $ QIR.findById issueReportId >>= fromMaybeM (IssueReportNotFound issueReportId.getId)
+      case (status, assignee) of
+        (Just justStatus, Just justAssignee) -> do
+          Esq.runTransaction $ QIR.updateStatusAssignee issueReportId (toDomainIssueStatus justStatus) justAssignee
+          Esq.runTransaction $ QC.create =<< mkIssueAssigneeUpdateComment justAssignee justUserId
+          pure Success
+        (Just justStatus, _) -> do
+          Esq.runTransaction $ QIR.updateStatus issueReportId (toDomainIssueStatus justStatus)
+          pure Success
+        (_, Just justAssignee) -> do
+          Esq.runTransaction $ QIR.updateAssignee issueReportId justAssignee
+          pure Success
+        (Nothing, Nothing) -> throwError $ InvalidRequest "Empty request, no fields to update."
+    Nothing ->
+      throwError $ InternalError "userId is not attatched to the request by dashboard."
   where
-    mkComment = do
+    mkIssueAssigneeUpdateComment justAssignee justUserId = do
       id <- generateGUID
       now <- getCurrentTime
       pure $
         DC.Comment
-          { createdAt = now,
-            ..
+          { id,
+            issueReportId,
+            authorId = cast justUserId,
+            comment = "Assignee Updated : " <> show justAssignee,
+            createdAt = now
           }
+
+issueAddComment :: ShortId DM.Merchant -> Id DIR.IssueReport -> Common.IssueAddCommentReq -> Flow APISuccess
+issueAddComment _merchantShortId issueReportId Common.IssueAddCommentReq {..} = do
+  case userId of
+    Just justUserId -> do
```

Comment:
userId <- justUserId & fromMaybe (InternalError "userId is not attatched to the request by dashboard.")